### PR TITLE
Player 4067

### PR DESCRIPTION
**Pull request by pilievOoyala**

### CODE REVIEW

File: js/components/controlBar.js [hunk 1]
```diff
@@ -553,6 +557,29 @@ var ControlBar = createReactClass({
         </div>
       ),
 
+      chromecast: function() {
```
Comment (aeng7) [inline js/components/controlBar.js]: Can we add a jsdoc?

**Comment by aeng7** [comment]
LGTM other than a minor doc comment

File: js/components/controlBar.js [hunk 2]
```diff
@@ -553,6 +557,29 @@ var ControlBar = createReactClass({
...
+        if (this.props.controller.state.chromecastAvailable !== true)
```
Comment (jpfranco) [inline js/components/controlBar.js]: Could we move this check to the `defaultItems` loop below for consistency?

File: js/components/controlBar.js [hunk 3]
```diff
@@ -553,6 +557,29 @@ var ControlBar = createReactClass({
...
+        if (this.props.controller.state.chromecastConnected === true)
+        {
+          castIcon = "chromecast-connected";
+        }
+        return (
+        <ControlButton
+          {...commonButtonProps}
+          key={CONSTANTS.CONTROL_BAR_KEYS.CHROMECAST}
+          focusId={CONSTANTS.CONTROL_BAR_KEYS.CHROMECAST}
+          ariaHidden={true}
```
Comment (jpfranco) [inline js/components/controlBar.js]: Is it deliberate to hide this button from screen readers?

File: js/controller.js [hunk 4]
```diff
@@ -1149,6 +1152,40 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
...
     },
 
+    /**
+     * Callback for event OO.EVENTS.CHROMECAST_AVAILABLE. It records that
+     * chromecasting is available on this player.
+     * @private
+     * @param  {string} event The event name OO.EVENTS.CHROMECAST_AVAILABLE
+     */
+    onChromecastAvailable: function(event) {
+      this.state.chromecastAvailable = true;
```
Comment (jpfranco) [inline js/controller.js]: Could we have a `chromecast` object in order to keep this grouped, such as:
```
chromecast: {
  isAvailable: false,
  isConnected: false
}
```

File: js/components/controlBar.js (hunk 3 above)
Comment (pilievOoyala) [reply js/components/controlBar.js]: No, I was copy pasting things till it worked. I'll change it

File: js/components/controlBar.js [hunk 5]
```diff
@@ -553,6 +557,31 @@ var ControlBar = createReactClass({
...
+      /**
+       * This function returns the chromecast button definition based on if
+       * it's connected to a chromecast or not
+       * @private
+       * @return {Object} The button definition
+       */
+      chromecast: function() {
+        let castIcon = "chromecast-disconnected";
+        if (this.props.controller.state.chromecast.isConnected === true)
+        {
```
Comment (jpfranco) [inline js/components/controlBar.js]: Sorry, minor nitpick but can we move this bracket to the line above? This is in order to be consistent with the styling on the rest of the file

File: js/components/controlBar.js [hunk 6]
```diff
@@ -553,6 +557,31 @@ var ControlBar = createReactClass({
...
+          castIcon = "chromecast-connected";
+        }
+        return (
+        <ControlButton
+          {...commonButtonProps}
+          key={CONSTANTS.CONTROL_BAR_KEYS.CHROMECAST}
+          focusId={CONSTANTS.CONTROL_BAR_KEYS.CHROMECAST}
+          ariaHidden={false}
```
Comment (jpfranco) [inline js/components/controlBar.js]: This only needed if set to `true`, I think it'd be better to remove it in this case

**Pull request merged**